sfs/util.py

```diff
@@ -466,3 +466,34 @@ def _count_walls_1d(a):
         order = order[max_order_mask, :]
         
     return xs, order
+
+
+def max_order_circular_harmonics(N, max_order):
```

Comment:
I think `max_order` should have a default of `None`.

But I'm not really sure if that function signature makes sense anyway ...

The "hidden" check for `None` seems a bit weird to me.

What about checking for `None` each time before use?

```python
if max_order is None:
    max_order = util.max_order_circular_harmonics(len(x0))
```

What do you think?